### DISCUSSION TITLE
What are all the "SpiralLinux" packages?

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#250] [Q&A] [answered]
I'm trying to "pin" all the packages tweaked/modified by Spiral so apt will prefer Spiral's packages and configs over Debian's.

Yo dev, do you have a list of packages you tweaked? Also, do you accept donations?

I don't care about themes. Just the core system components and system apps like Snapper.

**Comment by geckolinux** [comment, marked as answer]
Hi there, SpiralLinux does not modify any Debian packages, and there are no SpiralLinux repositories. This is an intentional design decision:
https://spirallinux.github.io/
> So a plethora of Debian-based distributions have arisen with the objective of adding a layer of polish, albeit with varying degrees of success. Most of these efforts depend on a single developer or a small group of developers with far fewer resources than the official Debian project. Many such projects create custom packages and supplemental package repositories that will cease to exist and leave their users stranded in the event that the project ever ceases to function. SpiralLinux, on the other hand, has been carefully designed to be completely dependent on the extensive development resources and excellent package infrastructure of the upstream Debian project. Debian itself provides a base system that is capable of being very user-friendly when properly configured. This is where SpiralLinux comes into play. Great effort has been expended in polishing the SpiralLinux default configuration for all the major desktop environments using the packages and mechanisms that Debian itself provides. So in effect, a SpiralLinux installation is actually a legitimate Debian installation that can be infinitely upgraded from the official Debian repositories while retaining its unique SpiralLinux configuration.

Config changes in SpiralLinux have been implemented so as to survive Debian upgrades. When upgrading SpiralLinux, it will behave exactly as if upgrading a personally customized Debian system-- Synaptic will show you if the new version is trying to overwrite any configs in `/etc/`  and let you choose how to handle it. In most cases it's best to keep the SpiralLinux config.

As for the configurations to the user's default environment, those originally get copied from `/etc/skel/` to `~/` during the user account creation process, as per standard Linux behavior. (These are usually hidden directories and `.dotfiles`). Those will not be changed during the upgrade process, but it is possible that a new version of a program or desktop environment might not support some of the previous user configurations or it might reset something in the user's config files, as would happen normally when upgrading a personally customized Debian system.

Regarding donations, thanks a ton for mentioning it, but I haven't set up any mechanism for receiving donations.

**Reply by birds-swim** [reply]
>Config changes in SpiralLinux have been implemented so as to survive Debian upgrades. 

Thank you, that is good news.

>When upgrading SpiralLinux, it will behave exactly as if upgrading a personally customized Debian system-- Synaptic will show you if the new version is trying to overwrite any configs in /etc/ and let you choose how to handle it. In most cases it's best to keep the SpiralLinux config.

Okay, so I will choose the "Keep" option. But in the future, as software changes, will you (as the dev) update any of these unique configs to keep up with the changes? How will YOUR updated configs be delievered to us? Do I need to worry about that at all? 

>As for the configurations to the user's default environment, those originally get copied from /etc/skel/ to ~/ during the user account creation process, as per standard Linux behavior. (These are usually hidden directories and .dotfiles).

So does that mean it'd be a good idea to backup these directories on a regular basis?

>Those will not be changed during the upgrade process, but it is possible that a new version of a program or desktop environment might not support some of the previous user configurations or it might reset something in the user's config files, as would happen normally when upgrading a personally customized Debian system.

This is what i'm worried about. What if the user adds a third-party repo (like Brave, for example)? How do I defend against that repo from making changes to Spiral's settings? I'm having a hard time asking this question and communicating effectively.

"How can I ask the user defend against you the developer if you fail to protect your changes?" I don't like how that's phrased; please forgive me. 

>Regarding donations, thanks a ton for mentioning it, but I haven't set up any mechanism for receiving donations.

PayPal is a popular choice, but please do consider non-PayPal sources as well.

**Comment by methanoid** [comment]
Can I suggest ONE thread rather than 3 to ask virtually the same questions? 

https://github.com/orgs/SpiralLinux/discussions/243
https://github.com/orgs/SpiralLinux/discussions/249

**Reply by birds-swim** [reply]
I'd delete them, but I don't have the permissions to.

I will edit them so that they will be empty. That's the closest to deletion I can get.
